code/modules/projectiles/guns/energy/special_eguns.dm

```diff
@@ -856,5 +856,54 @@
 			C.stop_tracking()
 	tracking_target_UID = null
 
+
+/obj/item/gun/energy/spikethrower //It's like the cyborg LMG, uses energy to make spikes
+	name = "\improper Vox spike thrower"
+	desc = "A vicious alien projectile weapon. Parts of it quiver gelatinously, as though the thing is insectile and alive."
+	icon = 'icons/obj/guns/projectile.dmi'
+	icon_state = "spikethrower"
+	item_state = "toxgun"
+	w_class = WEIGHT_CLASS_NORMAL
+	fire_sound_text = "a strange noise"
+	can_suppress = 0
+	burst_size = 2 // burst has to be stored here
+	can_charge = FALSE
+	selfcharge = TRUE
+	charge_delay = 10
+	restricted_species = list(/datum/species/vox)
+	ammo_type = list(/obj/item/ammo_casing/energy/spike)
+
+/obj/item/gun/energy/spikethrower/emp_act()
+	return
+
+/obj/item/ammo_casing/energy/spike
+	name = "alloy spike"
+	desc = "A broadhead spike made out of a weird silvery metal."
+	projectile_type = /obj/item/projectile/bullet/spike
+	muzzle_flash_effect = null
+	e_cost = 100
+	delay = 3 //and delay has to be stored here on energy guns
+	select_name = "spike"
+	fire_sound = 'sound/weapons/bladeslice.ogg'
+
+/obj/item/projectile/bullet/spike
+	name = "alloy spike"
+	desc = "It's about a foot of weird silvery metal with a wicked point."
+	damage = 25
+	knockdown = 2
+	armour_penetration_flat = 30
+	icon_state = "magspear"
+
+/obj/item/projectile/bullet/spike/on_hit(atom/target, blocked = 0)
+	if((blocked < 100) && ishuman(target))
+		var/mob/living/carbon/human/H = target
+		H.bleed(50)
+	..()
+
+/obj/item/gun/energy/examine(mob/user)
+	. = ..()
+	. += "<span class='notice'>This item's cell recharges on its own. Known to drive people mad by forcing them to wait for shots to recharge. Cannot be recharged in a recharger.</span>"
```

Comment:
```suggestion
	. += "<span class='notice'>This item's cell recharges on its own. Known to drive people mad by forcing them to wait for shots to recharge. Not compatible with rechargers.</span>"
```
To avoid repetition.